agentMET4FOF/metrological_streams.py

```diff
@@ -156,11 +163,20 @@ def __init__(
             quantity_units=quantity_units,
             misc=misc,
         )
+        self.value_unc = value_unc
+        self.time_unc = time_unc
         self.set_generator_function(
-            generator_function=self.sine_wave_function, sfreq=sfreq, F=F
+            generator_function=self._sine_wave_function,
+            uncertainty_generator=self._uncertainty_generator,
+            sfreq=sfreq,
+            F=F,
```

Comment:
It just occured to me that a more descriptive variable name than "F" might be needed here. Like "sine_freq" or "Fsin".

Reply:
Yes, I agree in general. The F is widely used across tutorials and in the original streams. We should either change it everywhere or leave it as it is, I would suggest.